web/middleware/etag.go

```diff
@@ -0,0 +1,49 @@
+package middleware
+
+import (
+	"crypto/md5"
```

Comment:
I know you're using `md5` as a non-cryptographic checksum, but it still makes me antsy to see a known-broken hash algorithm in new code